src/components/ReviewSummary/ReviewSummary.js

```diff
@@ -0,0 +1,104 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import React from 'react';
+import { connect } from 'react-redux';
+import styled from 'styled-components/native';
+import { getEnv } from 'configs/envConfig';
+import { CachedImage } from 'react-native-cached-image';
+import { BaseText, MediumText } from 'components/Typography';
+import { Spacing } from 'components/Layout';
+import CollectibleImage from 'components/CollectibleImage';
+import { formatAmount } from 'utils/common';
+import { getFormattedRate } from 'utils/assets';
+import { defaultFiatCurrency } from 'constants/assetsConstants';
+import type { Rates, Asset } from 'models/Asset';
+import type { Collectible } from 'models/Collectible';
+import type { RootReducerState } from 'reducers/rootReducer';
+
+
+type Props = {
+  amount?: number,
+  asset?: Asset,
+  collectible?: Collectible,
+  text: string,
+  rates: Rates,
+  baseFiatCurrency: ?string,
+};
+
+const Container = styled.View`
+  align-items: center;
+`;
+
+const TokenImage = styled(CachedImage)`
+  width: 64px;
+  height: 64px;
+`;
+
+const StyledCollectibleImage = styled(CollectibleImage)`
+  width: 192px;
+  height: 192px;
+`;
+
+const ReviewSummary = ({
+  asset, collectible, text, amount, rates, baseFiatCurrency,
+}: Props) => {
+  if (asset && amount) {
```

Comment:
Wouldn't it be simpler to have two separate components? The only shared part is the container style, and the variant for collectibles doesn't even have to be wrapped with `connect`. Then all relevant props could be marked as non-optional.

Unless having it this way makes more sense with the intended use -- then I'd only fix this condition so `amount` being equal to 0 does not cause the summary to disappear, just in case.